examples/go/vm-reconfig-via-tag/handler/handler.go

```diff
@@ -0,0 +1,564 @@
+}
+
+type vmConfig struct {
+	name    string
+	valBool bool
+	valInt  int
+}
+
+// Handle a function invocation
+func Handle(req handler.Request) (handler.Response, error) {
+	ctx := context.Background()
+	debug = setVerbosity()
+
+	// Load config every time, to ensure the most updated version is used.
+	cfg, err := loadConfig(cfgPath)
+	if err != nil {
+		return handlerResponseWithError("load vcconfig: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// The Mananged Object Reference for the VM that powered on.
+	vmMOR, err := eventVmMOR(req.Body)
+	if err != nil {
+		return handlerResponseWithError(
+			"retrieve VM object: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	if vsClt == nil {
+		// Save vSphere connection in a global variable.
+		vsClt, err = newClient(ctx, cfg)
+		if err != nil {
+			return handlerResponseWithError(
+				"connect to vSphere: %w",
+				http.StatusInternalServerError,
+				err,
+			)
+		}
+	}
+
+	// Signals wil cause vSphere APIs to log out.
+	once.Do(func() {
+		go handleSignal(ctx, vsClt)
+	})
+
+	// Check REST connection by getting tag objects.
+	tagObjs, err := vsClt.attachedTags(ctx, *vmMOR)
+	if err != nil {
+		// Retry once if not connected after creating new client.
+		if strings.Contains(err.Error(), "401 Unauthorized") {
+			if debug {
+				log.Println("401 Unauthorized. Retry connection to Govmomi.")
+			}
+
+			vsClt, err = newClient(ctx, cfg)
+		}
+
+		if err == nil {
+			tagObjs, err = vsClt.attachedTags(ctx, *vmMOR)
+		}
+	}
+
+	if err != nil {
+		return handlerResponseWithError(
+			"get tags attached to VM: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// Check SOAP connection by getting managed object VM.
+	moVM, err := vsClt.moVirtualMachine(ctx, *vmMOR)
+	if err != nil {
+		// Retry once after creating new client.
+		if strings.Contains(err.Error(), "NotAuthenticated") {
+			if debug {
+				log.Println("NotAuthenticated. Retry connection to VAPI.")
+			}
+
+			vsClt, err = newClient(ctx, cfg)
+		}
+
+		if err == nil {
+			moVM, err = vsClt.moVirtualMachine(ctx, *vmMOR)
+		}
+	}
+
+	if err != nil {
+		return handlerResponseWithError(
+			"retrieve current VM configs: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// Get the current configurations in a list from the VM object.
+	currCfgs := currentConfigs(ctx, moVM)
+
+	// Look for configurations that need to be applied to the VM.
+	newCfgs, err := vsClt.unappliedConfigs(ctx, tagObjs, currCfgs)
+	if err != nil {
+		return handlerResponseWithError(
+			"select only config tags: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	var task *object.Task
+
+	if len(newCfgs) > 0 {
+		task, err = vsClt.applyCfgs(ctx, moVM, *vmMOR, newCfgs)
+		if err != nil {
+			return handlerResponseWithError(
+				"apply config tags: %w",
+				http.StatusInternalServerError,
+				err,
+			)
+		}
+	}
+
+	message := appliedConfigsMessage(task, newCfgs)
+
+	return handler.Response{
```

Comment:
In your case, you're doing async task invocation, i.e. not `Wait()`, so status code should indicate (with 201/202)

Reply:
Oh, I didn't know that. Thanks for pointing that out. I'll change it to `202 Accepted`.